static/js/apps/eval_retrieval_generation/util.ts

```diff
@@ -63,61 +64,109 @@ export const processText = (text: string, calls?: DcCalls): string => {
-      let llmStat: string;
-      if (parts.length === 2) {
-        dcStat = parts[0];
-        llmStat = parts[1];
-      } else {
-        llmStat = parts[0];
-      }
+      const llmStat: string = parts.length === 2 ? parts[1] : parts[0];
 
       let innerHtml = "";
-      innerHtml += `<span class="llm-stat">${llmStat || LONG_SPACES}</span>`;
-      let annotationClasses = `annotation annotation-${callId}`;
+      innerHtml += `<span class="llm-stat">${
+        llmStat || LONG_SPACES
+      }<span class="material-icons-outlined">fact_check</span></span>`;
+      let annotationClasses = `annotation annotation-${callId} inline-stat`;
 
-      const hasDcStat = dcStat?.trim().length > 0;
-      if (hasDcStat) {
-        innerHtml += getTooltipHtml(dcStat, callId, calls);
-      } else {
+      const hasDcStat: boolean = calls && !!calls[callId]?.dcStat;
+      innerHtml += getTooltipHtml(hasDcStat, callId, footnotes, calls);
+      if (!hasDcStat) {
         annotationClasses += " annotation-no-dc-stat";
       }
 
       return `<span class="${annotationClasses}">${innerHtml}</span>`;
     }
   );
-  // Replace each link with the desired HTML format
-  return processedText.replace(
-    HTTP_PATTERN,
-    (match) => `<a href="${match}" target="_blank">Explore Page</a><br> `
-  );
+  return processedText;
 };
 
 function getTooltipHtml(
-  dcStat: string,
+  hasDcStat: boolean,
   callId: string,
+  footnotes: Map<string, Footnote>,
   calls?: DcCalls
 ): string {
-  let dcResponse = "";
-  if (calls) {
-    dcResponse = calls[callId]?.dcResponse || "";
+  let valueText = "",
+    linkHtml = "";
+  if (hasDcStat) {
+    const call = calls[callId];
+    valueText = call.dcStat;
+    if (call.dcResponse) {
+      valueText += ` - ${call.dcResponse}`;
+    }
+    if (valueText && footnotes.has(callId)) {
+      const footnote = footnotes.get(callId);
+      valueText += `; [REF] ${footnote.source} (${footnote.year})`;
+      linkHtml =
+        `<div class="dc-stat-tooltip-link">` +
+        `<a target="_blank" href="${footnote.link}">View more on Data Commons</a>` +
+        `</div>`;
+    }
+  } else {
+    valueText = "No reference found or available on Data Commons.";
   }
+
   return (
-    `<span class="dc-stat-tooltip">` +
-    `<span class="dc-stat-tooltip-value">${dcStat}</span>` +
-    `<br/>` +
-    `<span class="dc-stat-tooltip-label">${dcResponse}</span>` +
-    `</span>`
+    `<div class="dc-stat-tooltip">` +
+    `<div class="dc-stat-tooltip-title">Data Commons · Fact Check</div>` +
+    `<div class="dc-stat-tooltip-value">` +
+    valueText +
+    `</div>` +
+    linkHtml +
+    `</div>`
   );
 }
 
+const FOOTNOTE_REGEX =
+  /\[(\d+)\] - Per ([^\n]+), value was [^\n]+ in (\d+)\. See more at ([^\n]+)/g;
+
+interface Footnote {
+  source: string;
+  year: string;
+  link: string;
+}
+
+function extractFootnotes(text: string): Map<string, Footnote> {
+  const result = new Map();
+  const headerLocation = text.indexOf("Footnotes");
```

Comment:
nit: could we make use of the FOOTNOTE_HEADER_PATTERN constant here so incase footnote header changes, we can just change it in one place?

Reply:
oh good idea, done!